### PR TITLE
kubeadm: document alpha addon-installer init phase/feature-gate

### PR DESCRIPTION
**Pull request by stealthybox**
This is docs PR is for the feature described in https://github.com/kubernetes/enhancements/pull/1308 and implemented in https://github.com/kubernetes/kubernetes/pull/85224

It should not merge until both of these are merged at a minimum.

### CODE REVIEW

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 1]
```diff
@@ -153,6 +153,59 @@ Allowed customization are:
 Please note that the configuration field `kubernetesVersion` or the command line flag
 `--kubernetes-version` affect the version of the images.
 
+### Installing custom addons (Alpha) {#addon-installer}
+
+Kubeadm will normally install CoreDNS/kube-dns and kube-proxy using manifests built into the kubeadm binary.
+Using the `AddonInstaller` feature-gate (alpha), you can instruct kubeadm to install addons from other sources.
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]:
```suggestion
You can instruct kubeadm to install addons from other sources if you enable the `AddonInstaller` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
```

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 2]
```diff
@@ -153,6 +153,59 @@ Allowed customization are:
...
+Kubeadm will no longer use the in-binary manifests when using the `AddonInstaller` feature-gate.
+
+This feature is driven by a Component Config called the `addons.config.x-k8s.io/AddonInstallerConfiguration`.
+The `AddonInstallerConfiguration` can be specified in the kubeadm config-file.
+If the `AddonInstaller` feature-gate is specified, but a configuration is not supplied, kubeadm will default the API. It is planned to default to installing version appropriate CoreDNS+kube-proxy, but these addons are not yet implemented.
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: I recommend:
```suggestion
If you enable the `AddonInstaller` feature gate, but you don't specify a configuration, kubeadm uses its own default configuration.
```
([avoid making statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future))

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 3]
```diff
@@ -153,6 +153,59 @@ Allowed customization are:
...
+```shell
+kubeadm init --config cluster1-config.yaml
+```
+
+You may also invoke the addon-installer phase independently to apply the configuration without upgrading the rest of the cluster:
+```shell
+kubeadm init addons installer --config cluster1-config.yaml
+```
+
+The initial KEP for this feature can be found here: [kubeadm-addon-installer-KEP](TODO: link KEP merge location)
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: This isn't needed; I would omit it.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 4]
```diff
@@ -153,6 +153,59 @@ Allowed customization are:
 Please note that the configuration field `kubernetesVersion` or the command line flag
 `--kubernetes-version` affect the version of the images.
 
+### Installing custom addons (Alpha) {#addon-installer}
+
+Kubeadm will normally install CoreDNS/kube-dns and kube-proxy using manifests built into the kubeadm binary.
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]:
```suggestion
Normally, kubeadm deploys two [add-ons](/docs/concepts/cluster-administration/addons/), CoreDNS/kube-dns and kube-proxy, using manifests built into kubeadm itself.
```

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 5]
```diff
@@ -153,6 +153,59 @@ Allowed customization are:
 Please note that the configuration field `kubernetesVersion` or the command line flag
 `--kubernetes-version` affect the version of the images.
 
+### Installing custom addons (Alpha) {#addon-installer}
+
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: Add a [feature-state shortcode](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#feature-state) here.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 6]
```diff
@@ -153,6 +153,59 @@ Allowed customization are:
...
+Kubeadm will normally install CoreDNS/kube-dns and kube-proxy using manifests built into the kubeadm binary.
+Using the `AddonInstaller` feature-gate (alpha), you can instruct kubeadm to install addons from other sources.
+Kubeadm will no longer use the in-binary manifests when using the `AddonInstaller` feature-gate.
+
+This feature is driven by a Component Config called the `addons.config.x-k8s.io/AddonInstallerConfiguration`.
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]:
```suggestion
This feature is driven by a component configuration: `addons.config.x-k8s.io/AddonInstallerConfiguration`.
```

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 7]
```diff
@@ -153,6 +153,59 @@ Allowed customization are:
 Please note that the configuration field `kubernetesVersion` or the command line flag
 `--kubernetes-version` affect the version of the images.
 
+### Installing custom addons (Alpha) {#addon-installer}
+
+Kubeadm will normally install CoreDNS/kube-dns and kube-proxy using manifests built into the kubeadm binary.
+Using the `AddonInstaller` feature-gate (alpha), you can instruct kubeadm to install addons from other sources.
+Kubeadm will no longer use the in-binary manifests when using the `AddonInstaller` feature-gate.
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]:
```suggestion
If you enable the the `AddonInstaller` feature gate, kubeadm no longer uses its built-in add-on manifests.
```
How does this fit in with the explanation on line 164?

**Comment by sftim** [comment]
This should target https://github.com/kubernetes/website/tree/dev-1.18 once it exists

**Comment by sftim** [comment]
@stealthybox this PR should target https://github.com/kubernetes/website/tree/dev-1.18

I _think_ you can change this; if you need help to do that, it's available (eg reply to ask).

**Comment by sftim** [comment]
@stealthybox this PR should target https://github.com/kubernetes/website/tree/dev-1.18

I _think_ you can change this; if you need help to do that, it's available (eg reply to ask).

**Comment by sftim** [comment]
@stealthybox, are you happy to change the base branch for this pull request? (You might then want to do a rebase as well)

**Comment by kbhawkey** [comment]
@stealthybox 👋 
This PR has been open for some time. Are you actively working on the content changes?

**Comment by kbhawkey** [comment]
Ping @stealthybox , Do you have an update on this pull request?

**Comment by VineethReddy02** [comment]
Hi @stealthybox 
If this doc is targeted for 1.18 release. Can we require changes asap? As we are very close to the release deadline.

Thanks!

**Comment by VineethReddy02** [comment]
Hi @stealthybox 
As release 1.18 is tomorrow i.e 24th March. We are not accepting any more docs for the dev-1.18 release branch. Can you close this PR?

**Comment by sftim** [comment]
Kubernetes v1.18 was released. SIG Docs will be happy to look at a similar PR against _dev-1.19_ / future releases.

/close

**Pull request closed without merging**